sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -45,6 +46,27 @@
   private static final String ERROR_MESSAGE_INVALID_VALUE =
       " should be a ASCII string with a length not exceed " + MAX_LENGTH + " characters.";
   private static final Resource EMPTY = create(Attributes.empty());
+  private static final Resource TELEMETRY_SDK =
+      create(
+          Attributes.newBuilder()
+              .setAttribute("telemetry.sdk.name", "opentelemetry")
+              .setAttribute("telemetry.sdk.language", "java")
+              .setAttribute("telemetry.sdk.version", readVersion())
+              .build());
+
+  @Nullable
+  private static String readVersion() {
+
+    Properties properties = new Properties();
+    try {
+      properties.load(
+          Resource.class.getResourceAsStream("/io/opentelemetry/sdk/version.properties"));
+    } catch (Exception e) {
+      // we left the attribute empty
+      return null;
```

Comment:
Just noticed `null` value is allowed - @jkwatson does that seem OK? Wouldn't be surprised if it causes a NPE during export.

Either way, what about setting to `unknown`?

Reply:
My choice is based on the following [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes).

> Attribute values of null are considered to be not set and get discarded as if that SetAttribute call had never been made.

However, I noticed that the value is currently not discarded and, instead, a StringAttribute with value `null` is created. I was about to investigate a bit before opening an issue :wink:

Reply:
Related: to me, it sounds like the build itself should fail if the version is not properly detected.

Reply:
The build will make sure there is a version in the jar, but due to shading or similar, it may not make it into a user's app. I've seen a similar setup in Armeria cause NPE for users because of not accounting for the possibility before ><

Reply:
Yes, I would prefer "unknown" to be here, rather than possible nulls to confound an exporter that wasn't expecting it.

Reply:
Done in 6f113c0d